libflux/src/flux/semantic/builtins.rs

```diff
@@ -1,183 +1,189 @@
 use maplit::hashmap;
-use std::collections::hash_map;
 use std::collections::HashMap;
-use std::iter::Iterator;
-use std::vec::Vec;
+use std::collections::hash_map::Iter;
+use crate::semantic::import::Importer;
+use crate::semantic::parser::parse;
 
 pub struct Builtins<'a> {
-    pkgs: HashMap<&'a str, Node<'a>>,
+    pkgs: HashMap<&'a str, HashMap<&'a str, &'a str>>,
 }
 
 impl<'a> Builtins<'a> {
-    pub fn iter(&'a self) -> NodeIterator {
-        NodeIterator::new(self)
+    pub fn iter(&'a self) -> Iter<&'a str, HashMap<&'a str, &'a str>> {
+        self.pkgs.iter()
     }
-}
 
-enum Node<'a> {
-    Package(HashMap<&'a str, Node<'a>>),
-    Builtin(&'a str),
+    pub fn importer_for(&'a self, pkgpath: &str) -> impl Importer {
+        let mut h = HashMap::new();
+        match self.pkgs.get(pkgpath) {
+            Some(values) => {
+                for (name, expr) in values {
+                    h.insert(
+                        name.to_string(),
+                        parse(expr).unwrap(),
+                    );
+                }
+            }
+            _ => {},
+        }
+        return h
+    }
```

Comment:
This looks mostly correct, although I think we'll want to update any type variables with fresh ones before we insert into the importer. This can be accomplished using a combination of `max_tvar()` and methods defined in `fresh.rs`